assets/js/validateFormRegister.js

```diff
@@ -85,21 +88,26 @@ const validateFormRegister = (message) => {
 
   return isValid;
 }
-document.getElementById('formRegister').addEventListener('submit', (e) => {
+document.getElementById('formRegister').addEventListener('submit', async (e) => {
   e.preventDefault();
 
   const currentLang = getLanguage();
-
-  fetch(`/assets/locales/${currentLang}.json`)
-    .then(response => response.json())
-    .then(translations => {
-      const isValid = validateFormRegister(translations.alert);
-      if (isValid) {
-        successToast(toast);
-        setTimeout(() => {
-          successToast(toast, '-300px', '0');
-        }, 3000);
-      }
-    })
-    .catch(error => console.error('Lỗi tải bản dịch:', error));
+  try {
+    const response = await fetch(`/assets/locales/${currentLang}.json`);
+    const translations = await response.json();
+    data = {};
+    const isValid = validateFormRegister(translations.alert);
+    if (isValid) {
+      data.password = await hashPassword(data.password);
+      const userResponse = await post('users', data);
+      const payload = {
+        id: userResponse.id
+      };
+      const encodeToken = await createJWT(jwtHeader, payload, jwtSecret);
```

Comment:
```suggestion
      const userResponse = await post('users', data);
      const payload = {
        id: userResponse.id
      };
      const encodeToken = await createJWT(jwtHeader, payload, jwtSecret);
      
      data.password = await hashPassword(data.password);
```
Các biến sẽ đặt lên trước nha